src/main/java/uk/gov/cshr/controller/reset/ResetController.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.cshr.controller.reset;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import uk.gov.cshr.domain.Identity;
+import uk.gov.cshr.domain.Reset;
+import uk.gov.cshr.repository.IdentityRepository;
+import uk.gov.cshr.repository.ResetRepository;
+import uk.gov.cshr.service.ResetService;
+import uk.gov.service.notify.NotificationClientException;
+
+import javax.validation.Valid;
+
+@Controller
+@RequestMapping("/reset")
+public class ResetController {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResetController.class);
+
+    @Autowired
+    private ResetService resetService;
+
+    @Autowired
+    private ResetRepository resetRepository;
+
+    @Autowired
+    private IdentityRepository identityRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private ResetFormValidator resetFormValidator;
+
+    @GetMapping
+    public String reset() {
+        return "user-reset";
+    }
+
+    @PostMapping
+    public String requestReset(@RequestParam(value = "email") String email) throws Exception {
+        LOGGER.info("Requesting reset for {} ", email);
+
+        if (identityRepository.existsByEmail(email)) {
+            resetService.notifyForResetRequest(email);
+        }
+
+        return "user-checkEmail";
+    }
+
+    @GetMapping("/{code}")
+    public String loadResetForm(@PathVariable(value = "code") String code, RedirectAttributes redirectAttributes, Model model) {
+        LOGGER.info("User on reset screen with code {}", code);
+
+        Reset reset = resetRepository.findByCode(code);
+
+        if (!resetRepository.existsByCode(code)) {
+            LOGGER.info("{} reset code does not exist", code);
+            redirectAttributes.addFlashAttribute("status", "There was an error with your reset, please try again.");
+            return "redirect:/login";
```

Comment:
INFO - Is there a way to not redirect and kill the page right there with an error?